File: QID-1485-MVAboxcar/MVAboxcar.R

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# load data
x = read.table("carc.txt")

# parameter settings
k      = 0
l      = 0
m      = 0
us     = NULL
japan  = NULL
europe = NULL
M      = x[, 2]
C      = x[, 13]

for (i in 1:dim(x)[1]) {
    if (x[i, 13] == 1) {
        k = k + 1
        us[k] = x[i, 2]
    } else if (x[i, 13] == 2) {
        l = l + 1
        japan[l] = x[i, 2]
    } else if (x[i, 13] == 3) {
        m = m + 1
        europe[m] = x[i, 2]
    }
}

m1 = mean(us)
m2 = mean(japan)
m3 = mean(europe)

# plot
boxplot(us, japan, europe, axes = FALSE, frame = TRUE)
axis(side = 1, at = seq(1, 3), label = c("US", "JAPAN", "EU"))
axis(side = 2, at = seq(0, 50, 5), label = seq(0, 50, 5))
title("Car Data")
lines(c(0.6, 1.4), c(m1, m1), lty = "dotted", lwd = 1.2)
lines(c(1.6, 2.4), c(m2, m2), lty = "dotted", lwd = 1.2)
lines(c(2.6, 3.4), c(m3, m3), lty = "dotted", lwd = 1.2)

(five = quantile(x[, 2], c(0.025, 0.25, 0.5, 0.75, 0.975)))

File: QID-1485-MVAboxcar/MVAboxcar.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

x = pd.read_csv('carc.txt', sep="\t", header=None)

# parameter settings
k = 0
l = 0
m = 0
M = x[1]
C = x[12]

us = x[x[12] == 1][1]
japan = x[x[12] == 2][1]
europe = x[x[12] == 3][1]

m1 = us.mean()
m2 = japan.mean()
m3 = europe.mean()

fig, ax = plt.subplots(figsize = (10, 10))
ax.boxplot([us, japan, europe], labels = ["US", "JAPAN", "EU"], 
           meanline = True, showmeans = True, patch_artist=True, 
           boxprops=dict(facecolor = "lightgrey"), widths = 0.65, 
           medianprops=dict(color="black", linewidth = 2.5), 
           meanprops=dict(color="black"))
plt.title("Car Data", fontsize = 20)
plt.show()

pd.DataFrame(data = {"value": np.quantile(x[1], [0.025, 0.25, 0.5, 0.75, 0.975])}, 
             index = ["2.5%", "25%", "50%", "75%", "97.5%"]).T